src/main/java/net/mcreator/minecraft/MinecraftImageGenerator.java

```diff
@@ -325,29 +325,72 @@ public static BufferedImage generateBlastingPreviewPicture(Workspace workspace,
 		/**
 		 * <p>This method generates smithing recipe images.</p>
 		 *
+		 * @param template Template of the recipe.
 		 * @param input    Input of the recipe.
-		 * @param addition Addition of the recipe
+		 * @param addition Addition of the recipe.
 		 * @param result   Result of the recipe.
 		 * @return Returns the generated image.
 		 */
-		public static BufferedImage generateSmithingPreviewPicture(Workspace workspace, MItemBlock input,
-				MItemBlock addition, MItemBlock result) {
+		public static BufferedImage generateSmithingPreviewPicture(Workspace workspace, MItemBlock template,
+				MItemBlock input, MItemBlock addition, MItemBlock result) {
 			BufferedImage icon = new BufferedImage(28, 28, BufferedImage.TYPE_INT_ARGB);
+			int plusY = 11;
 			Graphics2D graphics2D = icon.createGraphics();
 			graphics2D.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
 			graphics2D.setColor(new Color(190, 190, 190, 65));
 
-			drawTwoSlotRecipe(graphics2D, workspace, input, addition);
+			if (template.isEmpty()) {
+				drawTwoSlotRecipe(graphics2D, workspace, input, addition);
+			} else {
+				plusY += 9;
+
+				//box 1
+				graphics2D.drawLine(10, 0, 17, 0);
```

Comment:
> Should I copy the code of the slot painting method or move `Graphics`'s origin position before calling it?

Reply:
Use offset variable/conditions, do not move origin

> > with new parameter and use conditions and math to draw different variations

Reply:
So current version is fine code-wise?

Reply:
Two boxes from before could/should be generated by drawTwoSlotRecipe. You can add offset parameter to drawTwoSlotRecipe

Reply:
✔️ Fixed